custom_components/opnsense/traffic_coordinator.py

```
@@ -0,0 +1,288 @@
+                RuntimeError("Live traffic stream ended without valid interface payload")
+            )
+            return False
+
+        # The stream ended after providing valid samples; mark unavailable so
+        # reconnect backoff is applied while preserving the last sample payload.
+        self.async_set_update_error(
+            RuntimeError("Live traffic stream ended and must be reconnected")
+        )
+        return has_valid_sample
+
+    def _consume_payload(self, payload: Mapping[str, Any]) -> bool:
```

Ruff (0.15.21)

[warning] Too many branches (14 > 12)

(PLR0912)